app/scripts/controllers/network/createJsonRpcClient.test.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * @jest-environment node
+ */
+import {
+  withMockedCommunications,
+  withClient,
+} from './provider-api-tests/helpers';
+import {
+  testsForRpcMethodNotHandledByMiddleware,
+  testsForRpcMethodAssumingNoBlockParam,
+  testsForRpcMethodsThatCheckForBlockHashInResponse,
+  testsForRpcMethodSupportingBlockParam,
+} from './provider-api-tests/shared-tests';
+
+describe('createJsonRpClient', () => {
+  // Ethereum JSON-RPC spec: <https://ethereum.github.io/execution-apis/api-documentation/>
+  const notHandledByMiddleware = [
+    { name: 'custom_rpc_method', numberOfParameters: 1 },
+    { name: 'eth_accounts', numberOfParameters: 0 },
+    { name: 'eth_coinbase', numberOfParameters: 0 },
+    { name: 'eth_feeHistory', numberOfParameters: 3 },
+    { name: 'eth_getFilterChanges', numberOfParameters: 1 },
+    { name: 'eth_getLogs', numberOfParameters: 1 },
+    { name: 'eth_getWork', numberOfParameters: 0 },
+    { name: 'eth_hashrate', numberOfParameters: 0 },
+    { name: 'eth_mining', numberOfParameters: 0 },
+    { name: 'eth_newBlockFilter', numberOfParameters: 0 },
+    { name: 'eth_newFilter', numberOfParameters: 1 },
+    { name: 'eth_newPendingTransactionFilter', numberOfParameters: 0 },
+    { name: 'eth_sendRawTransaction', numberOfParameters: 1 },
+    { name: 'eth_sendTransaction', numberOfParameters: 1 },
+    { name: 'eth_sign', numberOfParameters: 2 },
+    { name: 'eth_submitWork', numberOfParameters: 3 },
+    { name: 'eth_syncing', numberOfParameters: 0 },
+    { name: 'eth_uninstallFilter', numberOfParameters: 1 },
+    { name: 'eth_subscribe', numberOfParameters: 1 },
+    { name: 'eth_unsubscribe', numberOfParameters: 1 },
+    { name: 'net_listening', numberOfParameters: 0 },
+    { name: 'net_peerCount', numberOfParameters: 0 },
+    { name: 'parity_nextNonce', numberOfParameters: 1 },
```

Comment:
`parity_nextNonce` - I tested this method on a new infura api key and seems it doesn't work. Do we need to include this method, or any method with `parity` prefix?

Reply:
This is why I differentiated what was in Infura but not in the spec, or vice versa, because we seem to have legacy code in the middleware. So if we are explicitly handling this, I think we should test it, even if it's not supported in the spec, but we should note as such so that we know that it's legacy and that we should remove it in a future PR.

Reply:
ah okay. I did a rough search for things touching parity_nextNonce and came up with nothing, but to be fair I didnt search very exhaustively.